aten/src/ATen/FunctionalTensorWrapper.cpp

```diff
@@ -23,6 +23,19 @@ void FunctionalTensorWrapper::set_constructor_metadata() {
   storage_ = functional_storage;
   storage_access_should_throw_ = false;
   key_set_ = c10::DispatchKeySet(c10::DispatchKey::Functionalize) | value_.key_set();
+  // What's happening here? In general, functional wrappers *should* copy the keys from their inner tensor.
+  // We need this for the backend use case (LTC/XLA), where any functionalities that the backend uses,
+  // like autograd or autocast, need to run *directly* on the wrapper.
+  // However, all of the keys corresponding to functorch transforms should not be copied over.
+  // Functorch transforms all have their own wrapper tensors (e.g. BatchedTensorImpl) which expect
+  // to participate in the functorch transforms.
+  auto keys_to_not_copy =
+      c10::DispatchKeySet(c10::DispatchKeySet::FULL_AFTER, c10::DispatchKey::FuncTorchDynamicLayerFrontMode)
+    - c10::DispatchKeySet(c10::DispatchKeySet::FULL_AFTER, c10::DispatchKey::FuncTorchDynamicLayerBackMode)
```

Comment:
I think you want an explicit denylist with e.g. FuncTorchBatched, FuncTorchGradWrapper, and maybe the sparse keys(?) and the Lazy key (those are also 1:1 with their respective TensorImpl subclasses, right?)

The Autograd keys are also in between DynamicLayerFront and DynamicLayerBack and I'm not sure why you wanted to exclude them